Dynatrace/dynatrace-operator
# Fix race condition where code modules are not injected into ActiveGate pods

## Description

Injection of code modules into ActiveGate for self-monitoring purposes doesn't work without an ActiveGate pod restart because of a race condition and a wrong reconciliation loop ordering.

The connection info is not always written to the DynaKube status in timely enough that the code modules injection can pick it up when ActiveGate pods starts. Consequently, injection is postponed.
Additionally the app monitoring reconciliation was happening after ActiveGate deployment, so that it again was not ready when the ActiveGate pods start.

## How can this be tested?

* Deploy a cloud native DynaKube with the following feature flag applied:
```
feature.dynatrace.com/ignored-namespaces: '["^kube-.*", "openshift(-.*)?", "^gke-.*", "^gmp-.*" ]'
```
* Check if the code modules got injected into the ActiveGate pods
* Retry a few times and see if it works everytime

## Checklist
